aquadoggo/src/graphql/replication/query.rs

```diff
@@ -0,0 +1,421 @@
+        #[graphql(name = "publicKey", desc = "Public key of the entry author")]
+        public_key: scalars::PublicKey,
+    ) -> Result<EncodedEntryAndOperation> {
+        let store = ctx.data::<SqlStorage>()?;
+
+        let result = store
+            .get_entry_at_seq_num(&public_key.clone().into(), &log_id.into(), &seq_num.into())
+            .await?;
+
+        match result {
+            Some(inner) => Ok(EncodedEntryAndOperation::from(inner)),
+            None => Err(async_graphql::Error::new(format!(
+                "Entry with log id {}, sequence number {} and public key {} could not be found",
+                log_id, seq_num, public_key,
+            ))),
+        }
+    }
+
+    /// Get any entries that are newer than the provided sequence number for a given public key and
+    /// log id.
+    async fn entries_newer_than_seq_num<'a>(
+        &self,
+        ctx: &Context<'a>,
+        #[graphql(name = "logId", desc = "Log id of entries")] log_id: scalars::LogId,
+        #[graphql(name = "publicKey", desc = "Public key of the author")]
+        public_key: scalars::PublicKey,
+        #[graphql(
+            name = "seqNum",
+            desc = "Query entries starting from this sequence number"
+        )]
+        seq_num: Option<scalars::SeqNum>,
+        first: Option<i32>,
+        after: Option<String>,
+    ) -> Result<ConnectionResult> {
+        let store = ctx.data::<SqlStorage>()?;
+
+        query(
+            after,
+            None,
+            first,
+            None,
+            |after: Option<scalars::SeqNum>, _, first, _| async move {
+                // Add `seq_num` to the `after` cursor to get starting sequence number
+                let seq_num = seq_num.map(|seq| seq.as_u64()).unwrap_or_else(|| 0);
+                let start: u64 = seq_num + after.map(|a| a.as_u64()).unwrap_or_else(|| 0);
+
+                // `get_paginated_log_entries` is inclusive of seq_num. Whereas our seq_num should
+                // not be included. So we add 1 to the the seq_num we were passed
+                let start_seq_num = SeqNum::new(start + 1)?;
+
+                // Limit the maximum number of entries and set a default
+                let max_number_of_entries = first
+                    .map(|n| n.clamp(0, MAX_PAGINATION_SIZE))
+                    .unwrap_or(DEFAULT_PAGINATION_SIZE);
+
+                let edges = store
+                    .get_paginated_log_entries(
+                        &public_key.into(),
+                        &log_id.into(),
+                        &start_seq_num,
+                        max_number_of_entries,
+                    )
+                    .await?
+                    .into_iter()
+                    .map(|entry_and_operation| {
+                        // Unwrap as the entry was already validated before it entered the database
+                        //
+                        // @TODO: Is there a more efficient way than decoding the entry to access
```

Comment:
Yes! You can use `entry_and_operation.seq_num()` ;-p

Right now, that actually just decodes the entry under-the-hood... but that is because we have a rather inefficient implementation of `AsStorageEntry` which we can improve later with a small amount of effort.